stock_quant_unique_serial/models/stock_quant.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2023 Quartile Limited Odoo Community Association (OCA)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+
+    @api.constrains("product_id", "lot_id")
+    def _check_stock_quant_unique(self):
+        for record in self:
```

Comment:
@kanda999 Couln't you call the original Odoo method instead  (as it seems more complex than your code) ?

https://github.com/odoo/odoo/blob/16.0/addons/stock/models/stock_quant.py#L1011

Reply:
Have you ever thought about suggesting this to Odoo directly, maybe by adding a setting to enforce an error and not jsut a warning ?